Installer/AssetInstaller.php

```diff
@@ -58,13 +58,23 @@ public function supports($packageType)
     /**
      * {@inheritdoc}
      */
-    protected function getPackageBasePath(PackageInterface $package)
+    public function getInstallPath(PackageInterface $package)
     {
         $this->initializeVendorDir();
 
+        $targetDir = $package->getTargetDir();
+
         list(, $name) = explode('/', $package->getPrettyName(), 2);
 
-        return ($this->vendorDir ? $this->vendorDir.'/' : '').$name;
+        return ($this->vendorDir ? $this->vendorDir.'/' : '').$name.($targetDir ? '/'.$targetDir : '');
```

Comment:
Formatting should be fixed here. Missing spaces around `.`. The rest looks OK to me.